usecases/block_api.py

```diff
@@ -0,0 +1,22 @@
+from typing import Optional
+
+from elasticsearch import exceptions
+
+from gateways.block_api_gateway import BlockApiGateway
+
+
+class BlockApi:
+    def __init__(self, block_api_gateway: Optional[BlockApiGateway] = None) -> Optional[None]:
+        self.block_api_gateway = block_api_gateway or BlockApiGateway()
+
+    def get_best_chain_height(self) -> dict:
+        try:
+            return self.block_api_gateway.get_best_chain_height()
+        except exceptions.RequestError:
+            return {'error': 'bad_request', 'status': 400}
+        except exceptions.AuthorizationException:
+            return {'error': 'not_authorized', 'status': 403}
+        except exceptions.TransportError:
+            return {'error': 'elasticsearch_transport_error',  'status': 500}
+        except KeyError:
+            return {'error': 'elasticsearch_document_missing_information',  'status': 500}
```

Comment:
```suggestion
        except exceptions.RequestError:
            logger.error('ElasticSearch request error')
            raise ApiError('invalid_parameters')
        except exceptions.AuthorizationException:
            logger.error('ElasticSearch authorization error')
            raise ApiError('not_authorized')
        except (exceptions.TransportError, KeyError):
            logger.error('ElasticSearch transport error')
            raise ApiError('internal_error')
```

The `logger` is just as an example, we could log the errors to make it easier to check the source of the error.
But I think we should use the `ApiError`, so that we have a unified way to return api errors and not have each function implement it's own way to handle api errors

Reply:
Changed in the new commit. I've included the ApiError and the logger to enhance the traceability of the error